xtask/src/ci.rs

```diff
@@ -34,6 +37,10 @@ impl CiTask {
             }
         }
 
+        println!("Checking all Matrix Spec links point to same version...");
+        self.check_spec_links()?;
+        println!("All links to the Matrix Spec are consistent!");
```

Comment:
This won't actually run in CI since we don't run `cargo xtask ci`, we run `cargo xtask ci stable` in one job, `cargo xtask ci nightly` in another and `cargo xtask ci beta` in yet another one.

I think thematically it would make sense to move this into the `nightly` job since we're already checking other toolchain-unrelated things there. Probably makes sense to rename it because of that.